src/p_misc.c

```diff
@@ -476,118 +810,195 @@ prim_fork(PRIM_PROTOTYPE)
+ *
+ * @param player the player running the MUF program
+ * @param program the program being run
+ * @param mlev the effective MUCKER level
+ * @param pc the program counter pointer
+ * @param arg the argument stack
+ * @param top the top-most item of the stack
+ * @param fr the program frame
+ */
 void
 prim_pid(PRIM_PROTOTYPE)
 {
     CHECKOP(0);
     CHECKOFLOW(1);
+
     result = fr->pid;
     PushInt(result);
 }
 
+/**
+ * Implementation of MUF STATS
+ *
+ * Consumes a player dbref, and returns the number of owned objects as a
+ * series of integers: total, rooms, exits, things, programs, players, and
+ * garbage.  Requires MUCKER level 3.
+ *
+ * If the dbref is NOTHING, the integers represent all objects MUCK-wide.
+ *
+ * @param player the player running the MUF program
+ * @param program the program being run
+ * @param mlev the effective MUCKER level
+ * @param pc the program counter pointer
+ * @param arg the argument stack
+ * @param top the top-most item of the stack
+ * @param fr the program frame
+ */
 void
 prim_stats(PRIM_PROTOTYPE)
 {
     CHECKOP(1);
     oper1 = POP();
+
+    /**
+     * @TODO Should we allow anyone to summarize their own objects?
+     */
     if (mlev < 3)
-	abort_interp("Requires Mucker Level 3.");
+        abort_interp("Requires Mucker Level 3.");
+
     if (!valid_player(oper1) && (oper1->data.objref != NOTHING))
-	abort_interp("non-player argument (1)");
+        abort_interp("non-player argument (1)");
+
     ref = oper1->data.objref;
+
     CLEAR(oper1);
+
+    /**
+     * @TODO Why?
+     */
     nargs = 0;
+
+    /**
+     * @TODO Maybe use an array indexed by object type.  Too bad they aren't
+     *       defined in exactly this order.
+     */
     {
-	int rooms, exits, things, players, programs, garbage;
-
-	/* tmp, ref */
-	rooms = exits = things = players = programs = garbage = 0;
-	for (dbref i = 0; i < db_top; i++) {
-	    if (ref == NOTHING || OWNER(i) == ref) {
-		switch (Typeof(i)) {
-		case TYPE_ROOM:
-		    rooms++;
-		    break;
-		case TYPE_EXIT:
-		    exits++;
-		    break;
-		case TYPE_THING:
-		    things++;
-		    break;
-		case TYPE_PLAYER:
-		    players++;
-		    break;
-		case TYPE_PROGRAM:
-		    programs++;
-		    break;
-		case TYPE_GARBAGE:
-		    garbage++;
-		    break;
-		}
-	    }
-	}
-	ref = rooms + exits + things + players + programs + garbage;
-	CHECKOFLOW(7);
-	PushInt(ref);
-	PushInt(rooms);
-	PushInt(exits);
-	PushInt(things);
-	PushInt(programs);
-	PushInt(players);
-	PushInt(garbage);
-	/* push results */
+        int rooms, exits, things, players, programs, garbage;
+
+        /* tmp, ref */
+        rooms = exits = things = players = programs = garbage = 0;
+
+        for (dbref i = 0; i < db_top; i++) {
+            if (ref == NOTHING || OWNER(i) == ref) {
+                switch (Typeof(i)) {
+                    case TYPE_ROOM:
+                        rooms++;
+                        break;
+                    case TYPE_EXIT:
```

Comment:
To be technically correct (the best kind of correct) there should be a blank line between each 'case' block.  So like:

```
switch(Typeof(i)) {
    case TYPE_ROOM:
        rooms++;
        break;

    case TYPE_EXIT: /* and so forth */
```

Reply:
Done.